### ISSUE TITLE
Ensure compatibility with Google Site Kit

### ISSUE THREAD

**Issue by adamsilverstein** [WPGlobus/WPGlobus#94]
Hi! Adam here from the Google Site Kit team (https://sitekit.withgoogle.com). 

We recently had some users reporting issues when they were using WPGlobus and Site Kit at the same time. ([Github](https://github.com/google/site-kit-wp/issues/2118#issuecomment-703221014)), ([WordPress support forum](https://wordpress.org/support/topic/site-kit-asks-me-to-reconnect-all-the-time/page/2/#post-13485866)).

The underlying issue is that _Site Kit identifies site's by their URL_, as returned by WordPress's `home_url` function. When we detect that a site's URL has changed, we prompt (/force) the user to reconnect their site (going thru the auth process again). We do this to **help** users when they change their site URL for any reason (because otherwise, their connection is broken). 

As a consequence, when plugins filter the home_url this logic can cause issues as the user gets unexpectedly disconnected. WPGlobus does this in [filter__home_url](https://github.com/WPGlobus/WPGlobus/blob/28e1dec15dff98e8ab9dff67b0083b1c4a776f46/includes/class-wpglobus-filters.php#L441).

To improve compatibility in these types of situations, we added a `googlesitekit_canonical_home_url` filter in https://github.com/google/site-kit-wp/issues/2131. Site Kit uses the URL returned from this filter as our canonical URL, and as long as this URL doesn't change users will remain connected.

Using this filter, WPGlobus could add compatibility with Site Kit by returning the root site url of the site. 

Pseudo code:

```php
add_filter(
	'googlesitekit_canonical_home_url',
	function( $url ) {
		return get_option( 'home' ); // Return home url, or use stored value.
	}
);
```

I would be happy to open a PR with this change. Is this something you would consider adding to your plugin?

**Comment by tivnet** [comment]
Hi @adamsilverstein Adam,

Why don't you call `get_option( 'home' )` in your code directly? Just curious...

**Comment by adamsilverstein** [comment]
> Why don't you call get_option( 'home' ) in your code directly? Just curious...


@tivnet Good question! In general Site Kit wants to pick up when users change their site URL to keep their site connected to services with the correct URL. WordPress plugins sometimes effect this URL change with the `home_url` filter. If we used the option directly we would miss these changes.

For example, when a user installs an "HTTPS" plugin, their home url might change from `http://site.com` to `https://site.com`. Site Kit needs to detect this because we need to make sure the services the site is connected to (eg. Search Console, Analytics, AdSense) use the correct URL.  

In most cases, these types of URL changes are a "one time" event when the plugin is activated and disconnecting the user is expected: the user can just go thru the auth flow again _once_ to reconnect everything. Multilingual plugins are unusual in that they regularly change the url, including in wp-admin where Site Kit detects the change, causing users to be repeatedly disconnected. 

For this reason we are trying to work with each multilingual plugin where we see this issue (most are affected)  to have them introduce the compatibility using the filter we provide. 

Does WPGlobus already store the site's home/canonical URL?

If you want to test the issue, I was able to reproduce by following the ticket/reports. I created this mini-plugin to verify the fix: https://gist.github.com/adamsilverstein/199f3a4cfe8b4ccc881713a7ece05438

Thanks again for considering this change.
